### PR TITLE
add introduction about authentication changes

### PR DESCRIPTION
**Pull request by yiliuTo**
This pr is aimed to update the migration guide and reference doc of eh and sb binder,  to add the introduction about how to use security principals to retrieve connection strings.

### CODE REVIEW

**Review by saragluna — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 1]
```diff
@@ -1395,7 +1417,16 @@ spring:
               entity-type: queue #set as topic if needed
 ----
 
-To use the security principal to retrieve a connection string for authentication, taking service principal as example, configuration changes are listed as below:
+If you use other security principals instead of connection strings, in the legacy versions the application will firstly connect to Azure Resource Manager (ARM) with the provided security principal, and then
+retrieve the connection string of the specified namespace with ARM. In the end the application uses the retrieved connection string to authenticate to an Azure Service. In this way
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1421]: uses the retrieved connection string to **connect** an Azure **s**ervice

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 2]
```diff
@@ -1395,7 +1417,16 @@ spring:
               entity-type: queue #set as topic if needed
 ----
 
-To use the security principal to retrieve a connection string for authentication, taking service principal as example, configuration changes are listed as below:
+If you use other security principals instead of connection strings, in the legacy versions the application will firstly connect to Azure Resource Manager (ARM) with the provided security principal, and then
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1420]: in versions before 4.0

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 3]
```diff
@@ -1395,7 +1417,16 @@ spring:
...
+retrieve the connection string of the specified namespace with ARM. In the end the application uses the retrieved connection string to authenticate to an Azure Service. In this way
+the provided security principal should be granted with the link:https://docs.microsoft.com/azure/role-based-access-control/built-in-roles#contributor[Contributor] role to retrieve of the associated Azure Service namespace.
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1422]: to retrieve connection string of the associated Serivce Bus namespace.

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 4]
```diff
@@ -1395,7 +1417,16 @@ spring:
...
+
+For Azure Spring Cloud 4.0, we provide two ways of leveraging security principals for authentication. One is the legacy usage, the principals are used to
+connect to ARM and then retrieve connection string as credentials, where the `Contributor` role is required.
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1425]: remove the `as credentials`, connection strings contains credential information, but they != credentials

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 5]
```diff
@@ -1395,7 +1417,16 @@ spring:
               entity-type: queue #set as topic if needed
 ----
 
-To use the security principal to retrieve a connection string for authentication, taking service principal as example, configuration changes are listed as below:
+If you use other security principals instead of connection strings, in the legacy versions the application will firstly connect to Azure Resource Manager (ARM) with the provided security principal, and then
+retrieve the connection string of the specified namespace with ARM. In the end the application uses the retrieved connection string to authenticate to an Azure Service. In this way
+the provided security principal should be granted with the link:https://docs.microsoft.com/azure/role-based-access-control/built-in-roles#contributor[Contributor] role to retrieve of the associated Azure Service namespace.
+
+For Azure Spring Cloud 4.0, we provide two ways of leveraging security principals for authentication. One is the legacy usage, the principals are used to
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1424]: One is still using the principals to connect to ARM and retrieve the connection strings, where the `Contributor` role is required for the principals.

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 6]
```diff
@@ -1395,7 +1417,16 @@ spring:
...
+connect to ARM and then retrieve connection string as credentials, where the `Contributor` role is required.
+The other way leverages security principals to authenticate to Azure Active Directory and then connect to the Azure Service directly, in this case the `Contributor` role is not necessary anymore
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1426]: The other ~~way~~

Comment (saragluna) [reply docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1426]: the Azure service

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 7]
```diff
@@ -1431,8 +1462,10 @@ spring:
           resource-group: ${SERVICEBUS_RESOURCE_GROUP}
 ----
 
-To use the security principal for authentication directly, taking service principals as example, users could use below configurations, where the principal should
-be granted with necessary `Data` related roles for messaging operations.
+For authentication based on Azure Active Directory, security principals of the service principal and the manged identity are both supported, where the principal should
+be granted with necessary `Data` related roles for messaging operations. The configuration examples of the service principal and the manged identity are listed as below:
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1466]: You can also migrate to authenticate and authorize with Azure AD directly without making a detour to ARM. 
- With a service principal
- With a managed identity

**Review by saragluna — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 8]
```diff
@@ -1224,6 +1227,81 @@ spring:
...
+        tenant-id: ${TENANT_ID}
+        subscription-id: ${SUBSCRIPTION_ID}
+      eventhubs:
+        namespace: ${EVENTHUB_NAMESPACE}
+        resource:
+          resource-group: ${RESOURCE_GROUP}
+----
+
+You can also migrate to authenticate and authorize with Azure AD directly without making a detour to ARM. Security principals of the service principal and the manged identity are both supported, where the principal should
+be granted with necessary `Data` related roles for messaging operations. The configuration examples of the service principal and the manged identity are listed as below:
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:1273]: How about changing this `Security principals of the service principal and the manged identity are both supported, where the principal should
be granted with necessary `Data` related roles for messaging operations.` to `Make sure to grant the security principal necessary `Data` roles for messaging operations.` directly. The former expression will make people think we only support service principal and mi, but we do support user principals.

**Review by saragluna — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc (hunk 8 above)
Comment (yiliuTo) [reply docs/src/main/asciidoc/_migration-guide-for-4.0.adoc]: good idea, I ignored user principals as well

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**